clients/benchmarks/CMakeLists.txt

```diff
@@ -141,7 +139,7 @@ if( CUDA_FOUND )
 else( )
   # auto set in hip_common.h
   #target_compile_definitions( rocblas-bench PRIVATE __HIP_PLATFORM_HCC__ )
-  target_link_libraries( rocblas-bench PRIVATE ${HIPHCC_LOCATION} )
+  target_link_libraries( rocblas-bench PRIVATE hip::device )
```

Comment:
Does the clients have kernels in them? You should only link with `hip::device` when you have gpu kernels. 

Also, using `hip::host` shouldn't be needed either, as that should be provided by `roc::rocblas`(and if its not, then that should be considered a bug).

Reply:
The clients depend on rocblas, and rocblas may eventually become a static library.

Reply:
@amdkila do you know if rocblas-bench, rocblas-test or any samples have GPU kernels?

Reply:
rocblas-bench is a tool for GPU kernel benchmarking, so definitely yes. rocblas-test does CPU and GPU comparisons for accuracy, so I would think both hip::host and hip::device should be used. I do not know if hip::host is provided in roc::rocblas...but if it is, then @pfultz2 is right that there's no need to include hip::host. I do not know anything about rocBLAS samples, so I don't want to comment on whether GPU kernels are needed

Reply:
> The clients depend on rocblas, and rocblas may eventually become a static library.
 
Yes, even more so why rocblas should make sure its usage requirements are correct. The `hip::host` should be linked as `PUBLIC` or `INTERFACE` to rocblas, so that when you link against rocblas in the client, cmake will automatically link against `hip::host` for you as well.

Looking here:

https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/develop/library/src/CMakeLists.txt#L315

It appears rocblas does not link with `hip::host` correctly. It should add this line to the same cmake:

```cmake
target_link_libraries( rocblas PUBLIC hip::host )
```

Then the clients wont need to explicitly link against hip unless they are compiling device kernels. I assume this is not the case for clients since they are supposed to be built with vanilla gcc.